src/components/application_manager/src/rpc_handler_impl.cc

```diff
@@ -63,17 +65,56 @@ void RPCHandlerImpl::ProcessMessageFromMobile(
 #endif  // TELEMETRY_MONITOR
   smart_objects::SmartObjectSPtr so_from_mobile =
       std::make_shared<smart_objects::SmartObject>();
-
+  bool remove_unknown_parameters = true;
   DCHECK_OR_RETURN_VOID(so_from_mobile);
   if (!so_from_mobile) {
     LOG4CXX_ERROR(logger_, "Null pointer");
     return;
   }
 
-  if (!ConvertMessageToSO(*message, *so_from_mobile)) {
+  bool rpc_passing = app_manager_.GetAppServiceManager()
+                         .GetRPCPassingHandler()
+                         .CanHandleFunctionID(message->function_id());
+  if (app_manager_.GetRPCService().IsAppServiceRPC(
+          message->function_id(), commands::Command::SOURCE_MOBILE) ||
+      rpc_passing) {
+    LOG4CXX_DEBUG(logger_,
+                  "Allowing unknown parameters for request function "
+                      << message->function_id());
+    remove_unknown_parameters = false;
+  }
+
+  if (!ConvertMessageToSO(
+          *message, *so_from_mobile, remove_unknown_parameters)) {
     LOG4CXX_ERROR(logger_, "Cannot create smart object from message");
     return;
   }
+
+  if (rpc_passing) {
+    uint32_t correlation_id =
+        (*so_from_mobile)[strings::params][strings::correlation_id].asUInt();
+    int32_t message_type =
+        (*so_from_mobile)[strings::params][strings::message_type].asInt();
+    if (app_manager_.GetAppServiceManager()
+            .GetRPCPassingHandler()
+            .RPCPassThrough(*so_from_mobile)) {
+      LOG4CXX_ERROR(
```

Comment:
I wouldn't consider this an `error`, should probably change the log level

Reply:
Resolved in https://github.com/smartdevicelink/sdl_core/pull/2821/commits/d255c6004c831f4b208c1cec086a7088dafa7196